### PR TITLE
fix: update field ids to be backward compatible

### PR DESCRIPTION
**Pull request by kruskall**
https://github.com/elastic/apm-aggregation/pull/103/ broke backward compatibility because it changed the field ids.

This PR updates the field id to use to make it backward compatible with the old version.

### CODE REVIEW

**Review by lahsivjar — APPROVED** [review]
LGTM! Thanks

File: proto/aggregation.proto [hunk 1]
```diff
@@ -43,23 +43,26 @@ message KeyedTransactionMetrics {
...
-  string faas_name = 12;
-  string faas_version = 13;
-  string faas_trigger_type = 14;
+  uint32 faas_coldstart = 16;
+  string faas_id = 17;
+  string faas_name = 18;
+  string faas_version = 19;
+  string faas_trigger_type = 20;
+
+  // 21-29 are reserved for old fields
```
Comment (axw) [inline proto/aggregation.proto:64]: Should we use the `reserved` keyword? https://protobuf.dev/programming-guides/proto3/#fieldreserved

Also I suppose we should add `2-3`, `5`, `9-11` in there as well.

**Review by axw — APPROVED** [review]
thanks!

File: proto/aggregation.proto (hunk 1 above)
Comment (lahsivjar) [reply proto/aggregation.proto]: TIL too! This is nice!

**Pull request merged**